lib/identity/hostdata/config_builder.rb

```diff
@@ -77,6 +73,28 @@ def add(key, type: :string, allow_nil: false, enum: nil, options: {})
         @written_env[key] = converted_value.freeze
       end
 
+      def fetch_value_from_source(key, value)
+        if value.is_a?(Array) && value.length == 2 && value.first == 'env'
+          ENV.fetch(value[1])
+        elsif value.is_a?(Array) && value.length == 2 && value.first == 'secrets_manager'
+          secrets_client.get_secret_value(secret_id: value[1]).secret_string
+        elsif value.is_a?(Array)
+          raise "invalid configuration value for #{key}"
+        else
+          value
+        end
```

Comment:
WDYT of:
1. adding hash support
2. DRYing the the array checks:
```suggestion
        if value.is_a?(Array)
          type, name, *rest = value
          case type
          when 'env'
            ENV.fetch(name)
          when 'secrets_manager'
            secrets_client.get_secret_value(secret_id: name).secret_string
          else
            raise "invalid configuration value for #{key}"
          end
        else
          value
        end
```

Reply:
I think I'd rather tackle the hash separately from this to keep this focused on secrets manager support.